### ISSUE TITLE
[partition] improve pie chart labelling

### ISSUE THREAD

**Issue by markov00** [elastic/elastic-charts#2128]
Even if I strongly believe in the goodness of the current implementation of labels in pie chart, I believe we need to make an effort to improve the followings:
- find a way to show every label (inside or outside). Right now there are situation where outside labels are not represented.
- by default don't rotate the labels inside the slices, if doesn't fit put the label outside
- improve alignment of wrapped lines inside the slices (don't make them start from different positions where is not necessary)
<img width="488" alt="Screenshot 2023-08-03 at 15 21 54" src="https://github.com/elastic/elastic-charts/assets/1421091/bbe4d658-9dab-4c53-a4fa-2ce038ff63c1">

**Comment by gvnmagni** [comment]
Note for the design team: given the potential complexity of very crowded piecharts, we should consider the possibility of forcing the presence of a legend instead of finding a visual way to make all labels to be placed within the chart's area.

This is not a final solution (there are tons or pros and cons), it is just a consideration that we should keep in mind

**Comment by gvnmagni** [comment]
A first round of tests and draft brought us to this series of improvements that we should consider.

**1. Label style and structure**
In order to provide a better hierarchy of information the font weight have been switched, series name will be bolder while values will be lighter. 
Ideally, we would prefer using a two-lines structure but we know we have to deal with limited amount of space and, in case of need, there would be no problem in going on single line if that allows for the display of both elements (name and value).

![Export](https://github.com/elastic/elastic-charts/assets/18484576/c4c114ed-9d12-4f82-adc1-d08f3fc2985a)

**2. Colors**
Labels are currently full black which is not consistent with EUI while the EUI default color for texts is not accessible on top of all our charts colors. Therefore a compromise was needed, the new default color for labels would be `colors.title` [#1a1c21] which is accessible and not full black at the same time.

**3. Positioning**
Labels would be always oriented in the same classic direction instead of following the slices. We know this would cause few labels to disappear but we would like to reach a much better consistency and overall appearance. All labels would be at the same distance from the center of the pie (still to be decided how we define this distance and which is the reference point of these labels) and we would use a single-line version

![export 2](https://github.com/elastic/elastic-charts/assets/18484576/f17bb4ed-a9a8-4afc-aa78-b78cf62fbfb6)

**4. Labels disappearance**
We would like to switch from a "display all labels" approach to a "display all labels that could live comfortably" approach. This would cause few labels to disappear but one solution could be to force the presence of a legend in order to display all series values anyway. 
Note: we should consider what to do in case of report by users

**Comment by nickofthyme** [comment]
> We just have to calibrate that in order to balance the overall appearance

Exactly! And I think for slices over `𝜋`, the centroid would tend to pull into the center of the pie. So agree, there is a might be  several approaches based on the angle of the slice. We can discuss more when we start development.

**Comment by nickofthyme** [comment]
Fun gif Robert made years ago 😃 helps understand the current placement strategy.

![Image](https://github.com/user-attachments/assets/4d25967c-8a7d-4634-b35a-ab2e3c2704fe)

**Comment by gvnmagni** [comment]
Impressive 😳
